app/views/users/edit.haml

```diff
@@ -1,33 +1,4 @@
 %h1.title Edit user
 
-= form_for @user, url: roster_user_url(@roster, @user) do |f|
-  .input-group
-    = f.label :first_name
-    = f.text_field :first_name
-    = f.label :last_name
-    = f.text_field :last_name
-    = f.label :spire
-    = f.text_field :spire, size: 25, placeholder: '12345678@umass.edu'
-    = f.label :email
-    = f.text_field :email, size: 30
-    = f.label :phone
-    = f.text_field :phone, placeholder: '+14135450056'
-  .input-group
-    = f.label :rosters
-    = f.select :rosters,
-      options_from_collection_for_select(Roster.all, :id, :name,
-                                         selected: @user.rosters.pluck(:id)),
-      {}, multiple: true
-  .input-group
-    = f.label :reminders_enabled, 'Receive email reminders of upcoming assignments?'
-    = f.check_box :reminders_enabled
-  .input-group
-    = f.label :change_notifications_enabled, 'Receive notifications when your assignments are changed?'
-    = f.check_box :change_notifications_enabled
-    - if @current_user.admin_in? @roster
-      .input-group
-        = f.fields_for @user.membership_in(@roster) do |g|
-          = g.label :admin, "Admin in #{@roster.name}?"
-          = g.check_box :admin
-  .input-group
-    = f.submit 'Save'
+= render 'user', obj: @user, url: roster_user_url(@roster, @user),
+  roster: @user.rosters.pluck(:id), field: @user.membership_in(@roster)
```

Comment:
Any way to cut down on the number of locals? None of the logic is that complicated, and doing anything other than passing one (mayyyybe two) objects makes it so that interpreting the partial involves looking back at how it's invoked a bunch, which isn't ideal.

In particular if you can *just* use the instance variables then you don't need any locals at all.